### ISSUE TITLE
NullPointerException in native-image compilation with 1.0-RC8

### ISSUE THREAD

**Issue by peter-hofer** [oracle/graal#769]
_Originally posted by @Jiri-Kremser in https://github.com/oracle/graal/issues/582#issuecomment-434704242_

I am hitting the same issue with graalvm-ce-1.0.0-rc8.

```
git clone https://github.com/radanalyticsio/spark-operator
cd spark-operator && mvn clean install
native-image -H:Name=micro -J-Xmx4G -J-Xms4G -H:+JNI -H:EnableURLProtocols=https -H:+ReportUnsupportedElementsAtRuntime -jar ./target/spark-operator-*.jar

...
[spark-operator-0.2.1-SNAPSHOT:16909]    (compile):   7,453.18 ms
[spark-operator-0.2.1-SNAPSHOT:16909]      compile:  11,454.29 ms
fatal error: java.lang.NullPointerException
	at com.oracle.svm.hosted.code.CompileQueue.defaultCompileFunction(CompileQueue.java:896)
	at com.oracle.svm.hosted.code.CompileQueue.doCompile(CompileQueue.java:882)
	at com.oracle.svm.hosted.code.CompileQueue$CompileTask.run(CompileQueue.java:257)
	at com.oracle.graal.pointsto.util.CompletionExecutor.lambda$execute$0(CompletionExecutor.java:174)
	at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1402)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
```

**Comment by Hades32** [comment]
With assertions I get this:

```
Downloaded newer image for oracle/graalvm-ce:1.0.0-rc9
 ---> caf701c9e18f
Step 6/15 : EXPOSE 8080
 ---> Running in 28c92e27af05
Removing intermediate container 28c92e27af05
 ---> e00ceeef6feb
Step 7/15 : WORKDIR /app
 ---> Running in 7bc78e41b0fa
Removing intermediate container 7bc78e41b0fa
 ---> a3bd61934098
Step 8/15 : COPY --from=builder /app/target/cantina-*.jar cantina.jar
 ---> f14a30643a36
Step 9/15 : COPY --from=builder /app target
 ---> e06465d38fab
Step 10/15 : RUN java -cp cantina.jar io.micronaut.graal.reflect.GraalClassLoadingAnalyzer
 ---> Running in dab9c2ba7e99
12:55:50.757 [main] INFO  i.m.context.env.DefaultEnvironment - Established active environments: [cloud, gcp]
Graal Class Loading Analysis Enabled.
Graal Class Loading Analysis Enabled.
Writing reflect.json file to destination: target/reflect.json
Removing intermediate container dab9c2ba7e99
 ---> afad4fb8b6f7
Step 11/15 : RUN native-image --no-server -J-ea              --class-path cantina.jar 			 -H:ReflectionConfigurationFiles=target/reflect.json 			 -H:EnableURLProtocols=http,https 			 -H:IncludeResources="logback.xml|application.yml|META-INF/services/*.*" 			 -H:Name=cantina 			 -H:Class=cantina.Application 			 -H:+ReportUnsupportedElementsAtRuntime 			 -H:+AllowVMInspection 			 -H:-UseServiceLoaderFeature 			 --rerun-class-initialization-at-runtime='sun.security.jca.JCAUtil$CachedSecureRandomHolder,javax.net.ssl.SSLContext' 			 --delay-class-initialization-to-runtime=io.netty.handler.codec.http.HttpObjectEncoder,io.netty.handler.codec.http.websocketx.WebSocket00FrameEncoder,io.netty.handler.ssl.util.ThreadLocalInsecureRandom,com.sun.jndi.dns.DnsClient
 ---> Running in e53fc0320c90
[cantina:6]    classlist:   8,973.51 ms
[cantina:6]        (cap):   2,214.30 ms
[cantina:6]        setup:   6,024.44 ms
[cantina:6]     analysis:  58,138.04 ms
error: Error encountered while parsing com.oracle.svm.reflect.proxies.Proxy_io_reactivex_Single_constructor_.newInstance(java.lang.Object[]) 
Parsing context:
	parsing java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	parsing sun.net.www.protocol.http.Negotiator.getNegotiator(Negotiator.java:63)
	parsing sun.net.www.protocol.http.NegotiateAuthentication.isSupportedImpl(NegotiateAuthentication.java:138)
	parsing sun.net.www.protocol.http.NegotiateAuthentication.isSupported(NegotiateAuthentication.java:114)
	parsing sun.net.www.protocol.http.AuthenticationHeader.parse(AuthenticationHeader.java:220)
	parsing sun.net.www.protocol.http.AuthenticationHeader.<init>(AuthenticationHeader.java:143)
	parsing sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1644)
	parsing sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1492)
	parsing java.net.URL.openStream(URL.java:1045)
	parsing io.micronaut.core.io.service.SoftServiceLoader$ServiceLoaderIterator.hasNext(SoftServiceLoader.java:203)
	parsing io.micronaut.core.io.service.SoftServiceLoader$1.hasNext(SoftServiceLoader.java:144)
	parsing io.micronaut.context.DefaultBeanContext.resolveBeanDefinitionReferences(DefaultBeanContext.java:1002)
	parsing io.micronaut.context.DefaultApplicationContext.resolveBeanDefinitionReferences(DefaultApplicationContext.java:103)
	parsing io.micronaut.context.DefaultBeanContext.readAllBeanDefinitionClasses(DefaultBeanContext.java:1994)
	parsing io.micronaut.context.DefaultBeanContext.start(DefaultBeanContext.java:156)
	parsing io.micronaut.context.DefaultApplicationContext.start(DefaultApplicationContext.java:138)
	parsing io.micronaut.runtime.Micronaut.start(Micronaut.java:67)
	parsing cantina.Application.main(Application.kt:12)
	parsing com.oracle.svm.core.JavaMainWrapper.run(JavaMainWrapper.java:164)
	parsing com.oracle.svm.core.code.CEntryPointCallStubs.com_002eoracle_002esvm_002ecore_002eJavaMainWrapper_002erun_0028int_002corg_002egraalvm_002enativeimage_002ec_002etype_002eCCharPointerPointer_0029(generated:0)
Original error: java.lang.AssertionError: AnalysisType<io.reactivex.Single, allocated: false, inHeap: false, inTypeCheck: true>
	at com.oracle.graal.pointsto.meta.AnalysisType.registerAsAllocated(AnalysisType.java:526)
	at com.oracle.graal.pointsto.flow.MethodTypeFlowBuilder.registerUsedElements(MethodTypeFlowBuilder.java:235)
	at com.oracle.svm.hosted.analysis.flow.SVMMethodTypeFlowBuilder.registerUsedElements(SVMMethodTypeFlowBuilder.java:66)
	at com.oracle.graal.pointsto.flow.MethodTypeFlowBuilder.parse(MethodTypeFlowBuilder.java:212)
	at com.oracle.graal.pointsto.flow.MethodTypeFlowBuilder.apply(MethodTypeFlowBuilder.java:323)
	at com.oracle.graal.pointsto.flow.MethodTypeFlow.doParse(MethodTypeFlow.java:310)
	at com.oracle.graal.pointsto.flow.MethodTypeFlow.ensureParsed(MethodTypeFlow.java:300)
	at com.oracle.graal.pointsto.flow.MethodTypeFlow.addContext(MethodTypeFlow.java:107)
	at com.oracle.graal.pointsto.DefaultAnalysisPolicy$DefaultVirtualInvokeTypeFlow.onObservedUpdate(DefaultAnalysisPolicy.java:186)
	at com.oracle.graal.pointsto.flow.TypeFlow.notifyObservers(TypeFlow.java:347)
	at com.oracle.graal.pointsto.flow.TypeFlow.update(TypeFlow.java:389)
	at com.oracle.graal.pointsto.BigBang$2.run(BigBang.java:508)
	at com.oracle.graal.pointsto.util.CompletionExecutor.lambda$execute$0(CompletionExecutor.java:174)
	at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1402)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
```

**Comment by peter-hofer** [comment]
@Jiri-Kremser Thank you for your report, I managed to reproduce this issue with commit 457a076 of your repository (current HEAD fails due to a running Thread that is reachable in the image heap). The NullPointerException appears to be triggered by SLF4J using a `SecurityManager` to obtain the classes of the methods on the stack. I'll look into it.
@Hades32 Your issue seems to have a different cause. Do you have instructions on how to reproduce this?

**Comment by Hades32** [comment]
@peter-hofer I'll try to boil it down to a small sample sometime next week. But I'm pretty sure it only appeared after I activated the HTTPS URL protocol in the native-image tool...

**Comment by heziegl** [comment]
@Hades32 and @peter-hofer maybe same issue like #814 
same assertion even without httpS

**Comment by Hades32** [comment]
This assertion error definitely only happens after enabling HTTPS Schema:

```
error: Error encountered while parsing com.oracle.svm.reflect.proxies.Proxy_io_reactivex_Single_constructor_.newInstance(java.lang.Object[]) 
Parsing context:
	parsing java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	parsing sun.net.www.protocol.http.Negotiator.getNegotiator(Negotiator.java:63)
	parsing sun.net.www.protocol.http.NegotiateAuthentication.isSupportedImpl(NegotiateAuthentication.java:138)
	parsing sun.net.www.protocol.http.NegotiateAuthentication.isSupported(NegotiateAuthentication.java:114)
	parsing sun.net.www.protocol.http.AuthenticationHeader.parse(AuthenticationHeader.java:220)
	parsing sun.net.www.protocol.http.AuthenticationHeader.<init>(AuthenticationHeader.java:143)
	parsing sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1644)
	parsing sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1492)
	parsing java.net.URL.openStream(URL.java:1045)
	parsing io.micronaut.core.io.service.SoftServiceLoader$ServiceLoaderIterator.hasNext(SoftServiceLoader.java:203)
	parsing io.micronaut.core.io.service.SoftServiceLoader$1.hasNext(SoftServiceLoader.java:144)
	parsing io.micronaut.context.DefaultBeanContext.resolveBeanConfigurations(DefaultBeanContext.java:1018)
	parsing io.micronaut.context.DefaultApplicationContext.resolveBeanConfigurations(DefaultApplicationContext.java:95)
	parsing io.micronaut.context.DefaultBeanContext.readAllBeanConfigurations(DefaultBeanContext.java:1909)
	parsing io.micronaut.context.DefaultBeanContext.start(DefaultBeanContext.java:155)
	parsing io.micronaut.context.DefaultApplicationContext.start(DefaultApplicationContext.java:138)
	parsing io.micronaut.runtime.Micronaut.start(Micronaut.java:67)
	parsing cantina.Application.main(Application.kt:12)
	parsing com.oracle.svm.core.JavaMainWrapper.run(JavaMainWrapper.java:164)
	parsing com.oracle.svm.core.code.CEntryPointCallStubs.com_002eoracle_002esvm_002ecore_002eJavaMainWrapper_002erun_0028int_002corg_002egraalvm_002enativeimage_002ec_002etype_002eCCharPointerPointer_0029(generated:0)
Original error: java.lang.AssertionError: AnalysisType<io.reactivex.Single, allocated: false, inHeap: false, inTypeCheck: true>
	at com.oracle.graal.pointsto.meta.AnalysisType.registerAsAllocated(AnalysisType.java:526)
	at com.oracle.graal.pointsto.flow.MethodTypeFlowBuilder.registerUsedElements(MethodTypeFlowBuilder.java:235)
	at com.oracle.svm.hosted.analysis.flow.SVMMethodTypeFlowBuilder.registerUsedElements(SVMMethodTypeFlowBuilder.java:66)
	at com.oracle.graal.pointsto.flow.MethodTypeFlowBuilder.parse(MethodTypeFlowBuilder.java:212)
	at com.oracle.graal.pointsto.flow.MethodTypeFlowBuilder.apply(MethodTypeFlowBuilder.java:323)
	at com.oracle.graal.pointsto.flow.MethodTypeFlow.doParse(MethodTypeFlow.java:310)
	at com.oracle.graal.pointsto.flow.MethodTypeFlow.ensureParsed(MethodTypeFlow.java:300)
	at com.oracle.graal.pointsto.flow.MethodTypeFlow.addContext(MethodTypeFlow.java:107)
```

The reason might be that quite early in application startup a HTTPS call is happening. As far as I understood, Graal kind of starts an application during compilation, so maybe that's the problem...

**Comment by Hades32** [comment]
@cstancu so, not sure if it is the same as #814

**Comment by o-nix** [comment]
I added `--enable-https` and my build started to fail.

**Comment by peter-hofer** [comment]
The original issue by @Jiri-Kremser should be fixed with 4d4b6caee21 in 1.0-RC10.
